src/classes/RD2_OpportunityService_TEST.cls

```diff
@@ -186,6 +188,120 @@ private with sharing class RD2_OpportunityService_TEST {
+    */
+    @isTest
+    private static void shouldSetOppRecordTypeIdFromSettings() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Id rdSettingRecordTypeId = UTIL_RecordTypes_TEST.getRandomRecordTypeId(Opportunity.sObjectType);
+
+        Map<String, Object> rdSettingFieldMap = new Map<String, Object>{
+            'npe03__Record_Type__c' => rdSettingRecordTypeId
+        };
+        createRdSettingsWithFieldMap(rdSettingFieldMap);
+
+        Test.startTest();
+        createRecurringDonation();
+        Test.stopTest();
+
+        Opportunity installmentOpp = getOpportunities()[0];
+
+        System.assertNotEquals(null, installmentOpp, 'An opportunity should be created for each recurring donation record');
+        System.assertEquals(rdSettingRecordTypeId, installmentOpp.RecordTypeId,
+            'The record type should be configured according to the NPSP Opportunity Default Record Type');
+    }
+
+    /***
+    * @description The installment opportunity should have the default record type of the current user when the record
+    * type is not set in NPSP setting
+    */
+    @isTest
+    private static void shouldKeepDefaultOppRecordTypeIfRecordTypeSettingNotSet() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        createRdSettingsWithFieldMap(null);
+
+        Test.startTest();
+        createRecurringDonation();
+        Test.stopTest();
+
+        Opportunity installmentOpp = getOpportunities()[0];
+
+        Id defaultRecordTypeId = UTIL_RecordTypes_TEST.getDefaultRecordTypeId(Opportunity.sObjectType);
+
+        System.assertNotEquals(null, installmentOpp, 'An opportunity should be created for each recurring donation record');
+        System.assertEquals(defaultRecordTypeId, installmentOpp.RecordTypeId,
+            'The record type should be the default opp record type when NPSP default record type is null');
+    }
+
+    /***
+    * @description The installment opportunity should keep the original record type when the NPSP opp default record type
+    * is updated
+    */
+    @isTest
+    private static void shouldNotUpdateOppRecordTypeWhenRecordTypeSettingChange() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        createRdSettingsWithFieldMap(null);
+        npe03__Recurring_Donation__c rd = createRecurringDonation();
+
+        Opportunity installmentOpp = new TEST_OpportunityBuilder()
+            .withAmount(100)
+            .withAccount(getContact().AccountId)
+            .withClosedWonStage()
+            .withCloseDate(Date.today())
+            .withRecurringDonation(rd.Id)
+            .withCampaign(rd.npe03__Recurring_Donation_Campaign__c)
+            .build();
+        insert installmentOpp;
+
+        Test.startTest();
+        Id rdSettingRecordTypeId = UTIL_RecordTypes_TEST.getNonDefaultRecordTypeId(Opportunity.sObjectType);
```

Comment:
- Do we need `Test.start/stopTest()` in this test?
- If an org has zero or only one record type, will this test pass?